site/_posts/2024-11-27-fish-4.md

```diff
@@ -0,0 +1,78 @@
+---
+layout: post
+title: "fish-shell 4.0b and the Rust port"
+date: 2024-11-09
+categories: general
+---
+
+fish is a smart and user-friendly command line shell (like bash or zsh) with clever features that work out of the box, with nothing new to learn or configure.
+
+fish-shell is approaching its 4.0 milestone, the first release since porting from C++ to Rust. Today we are announcing an open beta, inviting all users to try out the upcoming 4.0 release.
+
+### **Try the fish-shell 4.0 beta at [fishshell.com](https://fishshell.com)**. View the [release notes here](https://fishshell.com/docs/current/relnotes.html).
+
```

Comment:
I think it's worth adding something here along the lines of "all existing fish scripts and commands should keep working, with the exception of the items noted in the release notes. In particular, the use of `?` as a wildcard is now disabled by default through a feature flag. Otherwise, if you notice something that isn't working as it should, that's a problem and we'd like to hear about it."